docs/managing-findings.md

```diff
@@ -0,0 +1,108 @@
+# Managing findings
+
+[TOC]
+
+# Findings
+
+A finding is the core unit of Semgrep's analysis. Findings are generated when a Semgrep rule matches a piece of code. After matching, a finding can make its way through 3 parts of the Semgrep ecosystem: [`semgrep`](https://github.com/returntocorp/semgrep), [`semgrep-action`](https://github.com/returntocorp/semgrep-action), and [semgrep.dev](https://semgrep.dev/).
+
+## Semgrep CLI
+
+Semgrep CLI findings are a specific rule matching a piece of code. Multiple rules can match the same piece of code, even if they are effectively the same rule. For example, consider the following rule and code snippet:
+
+```
+rules:
+- id: finding-test
+  pattern: $X == $X
+  message: Finding test 1
+  languages: [python]
+  severity: WARNING
+- id: finding-test
+  pattern: $X == $X
+  message: Finding test 2
+  languages: [python]
+  severity: WARNING
+```
+
+```
+print(1 == 1)
+```
+
+Running Semgrep CLI produces the following findings:
+
+```
+$ semgrep --quiet --config test.yaml test.py
+test.py
+severity:warning rule:finding-test: Finding test 1
+1:print(1 == 1)
+--------------------------------------------------------------------------------
+severity:warning rule:finding-test: Finding test 2
+1:print(1 == 1)
+```
+
+*For more information on writing rules, see [Rule syntax](writing-rules/rule-syntax.md).*
+
+## Semgrep Action
+
+Semgrep Action builds on Semgrep CLI findings to track the lifetime of an individual finding.
+
+A Semgrep Action finding is defined by a 4-tuple:
+
+```
+(rule ID, file path, syntactic context, index)
+```
+
+These pieces of state correspond to:
+
+1. `rule ID`: the rule's ID within the Semgrep ecosystem.
+1. `file path`: the filesystem path where the finding occurred.
+1. `syntactic context`: the lines of code corresponding to the finding.
+1. `index`: an index into identical findings within a file. This is used to disambiguate findings.
+
+!!! note
+    `syntactic context` is normalized by removing indentation, [`nosemgrep`](ignoring-findings.md#ignoring-findings-via-inline-comments) comments, and whitespace.
+
+These 4 pieces of information are then hashed and returned as the syntactic identifier: `syntactic_id`. This is how Semgrep Action uniquely identifies findings and tracks them across state transitions. Semgrep Action does not store or transmit code contents. The `syntactic context` is hashed using a one-way hashing function making it impossible to recover the original contents.
```

Comment:
```suggestion
These are hashed and returned as the syntactic identifier: `syntactic_id`. This is how Semgrep Action uniquely identifies findings and tracks them across state transitions. Semgrep Action does not store or transmit code contents. The `syntactic context` is hashed using a one-way hashing function making it impossible to recover the original contents.
```

Reply:
Thank you for reinforcing that Semgrep Action does not store or transmit code contents!